.travis.yml

```diff
@@ -29,15 +29,17 @@ before_cache:
 stages:
   - test
   - name: conda_dev_package
-    if: tag =~ ^v(\d+|\.)*[a-z]\d*$
+    if: tag =~ ^v(\d+|\.)+[a-z]\d+$
```

Comment:
I'm not objecting, but I'm not sure why the trailing digit is required.

Reply:
Background: I had a lot trouble with regexes on travis. The problem was figuring out what syntax to use, and the fact that it's not the same in every part of the file (plus it's not all documented...plus travis condition matching has changed over the lifetime of pyctdev). I generally had to experiment to get something to work, with feedback only after a build. So, the regexes became a bit of a mess.

Originally, I needed the same regex in two parts of the file, but had to write them in two different ways. However, over time, I've settled on a simpler approach where the regexes only appear in one place (as in this .travis file). So the only thing necessary to understand/get right now are these regexes at the start of the file.

So finally I can get onto your question! 

> I'm not sure why the trailing digit is required

"python versioning" normalizes `v1.2.3a` to `v1.2.3a0`, so I think it makes sense for us to include the trailing digit:

```python
>>> from packaging import version
>>> version.parse("1.2.3a")
<Version('1.2.3a0')>
```

However, that doesn't really answer your question, because I think the regexes could be simplified. Right now we have:

dev:
```
    if: tag =~ ^v(\d+|\.)+[a-z]\d+$
```

release:
```
    if: tag =~ ^v(\d+|\.)+[^a-z]\d+$
```

They're confusing to read, but work (i.e. have been tested and are in use). I think I could write down clearer versions right now...but I don't want to mess with it. If someone does want to simplify them, great, but it would require testing that they actually work, which is a pain. So, I've just been including a comment to that effect.


-----

I think ideally everything would be in pyctdev so there's no knowledge spread around .travis and .appveyor files. E.g. pyctdev could decide what a tag means and do the appropriate thing. However, then you'd lose certain features of the build systems (e.g. meaningful presentation, parallelization), so it's a compromise.

Reply:
Ok.